Support https in monetization support via elasticsearch

### Description

When monetization support is enabled via elastic search, it fails to get usage data from elastic search when using https.

Throws the following error:

```
[2025-09-23 11:38:22,731] ERROR - MonetizationUsagePublishAgent Failed to publish monetization usage to billing Engine
org.wso2.carbon.apimgt.api.MonetizationException: Error occurred while executing data retrieval from Elasticsearch
	at org.wso2.apim.monetization.impl.StripeMonetizationImpl.getUsageDataFromElasticsearch(StripeMonetizationImpl.java:1105) ~[org.wso2.apim.monetization.impl-1.5.0.jar:?]
	at org.wso2.apim.monetization.impl.StripeMonetizationImpl.publishMonetizationUsageRecords(StripeMonetizationImpl.java:718) ~[org.wso2.apim.monetization.impl-1.5.0.jar:?]
	at org.wso2.carbon.apimgt.impl.monetization.MonetizationUsagePublishAgent.run_aroundBody0(MonetizationUsagePublishAgent.java:62) ~[org.wso2.carbon.apimgt.impl_9.32.75.jar:?]
	at org.wso2.carbon.apimgt.impl.monetization.MonetizationUsagePublishAgent.run(MonetizationUsagePublishAgent.java:1) ~[org.wso2.carbon.apimgt.impl_9.32.75.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: org.apache.http.ConnectionClosedException: Connection is closed
	at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:924) ~[org.wso2.apim.monetization.impl-1.5.0.jar:?]
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:304) ~[org.wso2.apim.monetization.impl-1.5.0.jar:?]
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:292) ~[org.wso2.apim.monetization.impl-1.5.0.jar:?]
	at co.elastic.clients.transport.rest_client.RestClientHttpClient.performRequest(RestClientHttpClient.java:91) ~[org.wso2.apim.monetization.impl-1.5.0.jar:?]
	at co.elastic.clients.transport.ElasticsearchTransportBase.performRequest(ElasticsearchTransportBase.java:144) ~[org.wso2.apim.monetization.impl-1.5.0.jar:?]
	at co.elastic.clients.elasticsearch.ElasticsearchClient.search(ElasticsearchClient.java:1923) ~[org.wso2.apim.monetization.impl-1.5.0.jar:?]
	at org.wso2.apim.monetization.impl.StripeMonetizationImpl.getUsageDataFromElasticsearch(StripeMonetizationImpl.java:1103) ~[org.wso2.apim.monetization.impl-1.5.0.jar:?]
	... 6 more
Caused by: org.apache.http.ConnectionClosedException: Connection is closed
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.endOfInput(HttpAsyncRequestExecutor.java:356) ~[httpcore-nio_4.4.16.wso2v4.jar:?]
	at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:261) ~[httpcore-nio_4.4.16.wso2v4.jar:?]
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:87) ~[?:?]
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:40) ~[?:?]
	at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114) ~[httpcore-nio_4.4.16.wso2v4.jar:?]
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162) ~[httpcore-nio_4.4.16.wso2v4.jar:?]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337) ~[httpcore-nio_4.4.16.wso2v4.jar:?]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315) ~[httpcore-nio_4.4.16.wso2v4.jar:?]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276) ~[httpcore-nio_4.4.16.wso2v4.jar:?]
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104) ~[httpcore-nio_4.4.16.wso2v4.jar:?]
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591) ~[httpcore-nio_4.4.16.wso2v4.jar:?]
	... 1 more
```

The following error is thrown from `elasticsearch` side:

```
[2025-09-23T11:38:22,723][WARN ][o.e.h.n.Netty4HttpServerTransport] [User.local] received plaintext http traffic on an https channel, closing connection Netty4HttpChannel{localAddress=/127.0.0.1:9200, remoteAddress=/127.0.0.1:55994}
```


### Steps to Reproduce

N/A

### Version

4.6.0

### Environment Details (with versions)

_No response_

--- a/pom.xml
+++ b/pom.xml
@@ -20,6 +20,12 @@
             <groupId>org.wso2.carbon.apimgt</groupId>
             <artifactId>org.wso2.carbon.apimgt.impl</artifactId>
             <version>${carbon.apimgt.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.wso2.am.analytics.publisher</groupId>
+                    <artifactId>org.wso2.am.analytics.publisher.client</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.wso2.carbon.apimgt</groupId>
@@ -50,6 +56,12 @@
             <groupId>org.elasticsearch.client</groupId>
             <artifactId>elasticsearch-rest-client</artifactId>
             <version>${elastic.rest.client.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.httpcomponents</groupId>
+                    <artifactId>httpasyncclient</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>io.opentelemetry</groupId>
@@ -70,7 +82,8 @@
     </dependencies>
 
     <properties>
-        <carbon.apimgt.version>9.29.119</carbon.apimgt.version>
+<!--        TODO: UPDATE VERSION-->
+        <carbon.apimgt.version>9.32.119-SNAPSHOT</carbon.apimgt.version>
         <commons.lang.version>2.6</commons.lang.version>
         <stripe.version>9.8.0</stripe.version>
         <elastic.client.version>8.13.1</elastic.client.version>

--- a/pom.xml
+++ b/pom.xml
@@ -82,8 +82,7 @@
     </dependencies>
 
     <properties>
-<!--        TODO: UPDATE VERSION-->
-        <carbon.apimgt.version>9.32.119-SNAPSHOT</carbon.apimgt.version>
+        <carbon.apimgt.version>9.32.130</carbon.apimgt.version>
         <commons.lang.version>2.6</commons.lang.version>
         <stripe.version>9.8.0</stripe.version>
         <elastic.client.version>8.13.1</elastic.client.version>

--- a/src/main/java/org/wso2/apim/monetization/impl/StripeMonetizationConstants.java
+++ b/src/main/java/org/wso2/apim/monetization/impl/StripeMonetizationConstants.java
@@ -162,6 +162,8 @@ public class StripeMonetizationConstants {
     public static final String ELK_TENANT_DOMAIN = "apiCreatorTenantDomain.keyword";
     public static final String APPLICATION_ID_COLUMN = "applicationId";
     public static final String ELK_APPLICATION_ID_COLUMN = "applicationId.keyword";
+    public static final String HTTP_PROTOCOL = "http";
+    public static final String HTTPS_PROTOCOL = "https";
 
     public static final String MONETIZATION_PROXY_ENABLE_CONFIG = "Monetization.ProxyEnabled";
 }

--- a/src/main/java/org/wso2/apim/monetization/impl/StripeMonetizationImpl.java
+++ b/src/main/java/org/wso2/apim/monetization/impl/StripeMonetizationImpl.java
@@ -724,7 +724,8 @@ public class StripeMonetizationImpl implements Monetization {
         String formattedFromDate = fromDate.concat(StripeMonetizationConstants.TIMEZONE_FORMAT);
 
         if (config.getFirstProperty("Analytics.Type") != null
-                && !config.getFirstProperty("Analytics.Type").equals("")) {
+                && !config.getFirstProperty("Analytics.Type").isEmpty()
+                && !config.getFirstProperty("Analytics.Type").equals("choreo")) {
             SearchResponse<Object> searchResponse = getUsageDataFromElasticsearch(fromDate, toDate);
 
             if (log.isDebugEnabled()) {

--- a/src/main/java/org/wso2/apim/monetization/impl/StripeMonetizationImpl.java
+++ b/src/main/java/org/wso2/apim/monetization/impl/StripeMonetizationImpl.java
@@ -53,7 +53,9 @@ import org.apache.http.HttpHost;
 import org.apache.http.auth.AuthScope;
 import org.apache.http.auth.UsernamePasswordCredentials;
 import org.apache.http.client.CredentialsProvider;
+import org.apache.http.conn.ssl.DefaultHostnameVerifier;
 import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.ssl.SSLContexts;
 import org.elasticsearch.client.RestClient;
 import org.json.simple.JSONArray;
 import org.json.simple.JSONObject;
@@ -103,6 +105,7 @@ import org.wso2.carbon.user.api.Tenant;
 import org.wso2.carbon.user.api.UserStoreException;
 import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
 
+import javax.net.ssl.SSLContext;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.sql.Connection;
@@ -128,6 +131,8 @@ import static org.wso2.apim.monetization.impl.StripeMonetizationConstants.ELK_TE
 import static org.wso2.apim.monetization.impl.StripeMonetizationConstants.PRODUCTS;
 import static org.wso2.apim.monetization.impl.StripeMonetizationConstants.REQUEST_TIMESTAMP_COLUMN;
 import static org.wso2.apim.monetization.impl.StripeMonetizationConstants.TENANT_DOMAIN_COL;
+import static org.wso2.apim.monetization.impl.StripeMonetizationConstants.HTTP_PROTOCOL;
+import static org.wso2.apim.monetization.impl.StripeMonetizationConstants.HTTPS_PROTOCOL;
 
 /**
  * This class is used to implement stripe based monetization
@@ -1061,12 +1066,30 @@ public class StripeMonetizationImpl implements Monetization {
             analyticsIndex = DEFAULT_ELK_ANALYTICS_INDEX;
         }
         int port = config.getMonetizationConfigurationDto().getAnalyticsPort();
+        String protocol = config.getMonetizationConfigurationDto().getAnalyticsProtocol();
+        if (protocol == null) {
+            protocol = HTTP_PROTOCOL;
+        }
+
         List<JSONArray> tenantDomainsAndAPIs = getMonetizedAPIIdsAndTenantDomains();
         JSONArray tenants = tenantDomainsAndAPIs.get(0);
         JSONArray monetizedAPIs = tenantDomainsAndAPIs.get(1);
         final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
         credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username,
                 new String(password, StandardCharsets.UTF_8)));
+
+        final SSLContext sslContext;
+        if (HTTPS_PROTOCOL.equalsIgnoreCase(protocol)) {
+            try {
+                sslContext = SSLContexts.custom()
+                        .loadTrustMaterial(ServiceReferenceHolder.getInstance().getTrustStore(), null)
+                        .build();
+            } catch (Exception e) {
+                throw new MonetizationException("Failed to initialize SSL context for Elasticsearch connection", e);
+            }
+        } else {
+            sslContext = null;
+        }
 
         if (tenantDomainsAndAPIs.size() == 2) {
             List<FieldValue> tenantList = new ArrayList<>();
@@ -1077,9 +1100,15 @@ public class StripeMonetizationImpl implements Monetization {
             for (Object api : monetizedAPIs) {
                 monetizedAPIsList.add(new FieldValue.Builder().stringValue((String) api).build());
             }
-            try (RestClient restClient = RestClient.builder(new HttpHost(hostname, port))
-                    .setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder
-                            .setDefaultCredentialsProvider(credentialsProvider)).build()) {
+            try (RestClient restClient = RestClient.builder(new HttpHost(hostname, port, protocol))
+                    .setHttpClientConfigCallback(httpClientBuilder -> {
+                        if (sslContext != null) {
+                            httpClientBuilder.setSSLContext(sslContext)
+                                    .setSSLHostnameVerifier(new DefaultHostnameVerifier());
+                        }
+                        return httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+                    }).build()) {
+
                 ElasticsearchTransport transport = new RestClientTransport(restClient, new JacksonJsonpMapper());
                 ElasticsearchClient elasticsearchClient = new ElasticsearchClient(transport);
 
@@ -1110,9 +1139,15 @@ public class StripeMonetizationImpl implements Monetization {
                 throw new MonetizationException("Error occurred while executing data retrieval from Elasticsearch", e);
             }
         } else {
-            try (RestClient restClient = RestClient.builder(new HttpHost(hostname, port))
-                    .setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder
-                            .setDefaultCredentialsProvider(credentialsProvider)).build()) {
+            try (RestClient restClient = RestClient.builder(new HttpHost(hostname, port, protocol))
+                    .setHttpClientConfigCallback(httpClientBuilder -> {
+                        if (sslContext != null) {
+                            httpClientBuilder.setSSLContext(sslContext)
+                                    .setSSLHostnameVerifier(new DefaultHostnameVerifier());
+                        }
+                        return httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+                    }).build()) {
+
                 ElasticsearchTransport transport = new RestClientTransport(restClient, new JacksonJsonpMapper());
                 ElasticsearchClient elasticsearchClient = new ElasticsearchClient(transport);
                 Query query = RangeQuery.of(r -> r.field(REQUEST_TIMESTAMP_COLUMN)
@@ -1658,7 +1693,7 @@ public class StripeMonetizationImpl implements Monetization {
         UserContext userCtx = new UserContext(username, org, properties, roles);
         try {
             PublisherAPISearchResult searchAPIs = apiPersistenceInstance.searchAPIsForPublisher(org, "", 0,
-                    Integer.MAX_VALUE, userCtx, null, null);
+                    Integer.MAX_VALUE, userCtx);
 
             if (searchAPIs != null) {
                 List<PublisherAPIInfo> list = searchAPIs.getPublisherAPIInfoList();

